test/join.jl

```diff
@@ -1,8 +1,22 @@
 module TestJoin
     using Compat, Compat.Test, DataFrames
     using DataFrames: similar_missing
+    using TimeZones
+    import Base.==
+    import Base.hash
+    import Base.isequal
+
     const ≅ = isequal
 
+    type TestSum
```

Comment:
DataFrames uses Julia 0.6 or newer, so `type` should instead be `mutable struct` (or just `struct` if you don't need to be able to change the fields after construction).